lib/date_time/date_time_internal.h

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef DATE_TIME_INTERNAL_H_
+#define DATE_TIME_INTERNAL_H_
+
+void date_time_store(int64_t curr_time_ms, enum date_time_evt_type time_source);
+int date_time_current_check(void);
```

Comment:
Instead of `date_time_internal.h`, we could have completely separate module for some of the core functions such as these, and maybe most of the static functions from `date_time.c`.
It's about how far we take the modularization so that it better serves `date_time` library, which I don't expect to grow significantly.
@simensrostad Let me know your thoughts.

Reply:
The library looks much cleaner now. And as you mention in your comment it's probably not going to grow any time soon. I don't think `date_time_internal.h` is needed.

Reply:
Right now we need `date_time_internal.h` or some other header because it has these two function declarations that `date_time_modem.c` needs.
Alternatively, we could have `date_time_core.c` and `date_time_core.h` which would have these two functions and maybe also something additional from current `date_time.c` file.

Reply:
Ah, right. I didn't catch that. In that regard, I think we should abstract those into a separate c/h file combo.

How about `date_time_helpers.c` / `date_time_helpers.h` ?

Reply:
I can split `date_time.c` into `date_time.c` and `date_time_helpers.c` but it's not going to be completely clean because most of the static variables from current `date_time.c` would need to be revealed in `date_time_helpers.c` too. In addition to `date_time_store` and `date_time_current_check`, you would need to take functions they call (`date_time_schedule_update` and `date_time_notify_event`), which would be actually nice unless you had then all static variables in use in helpers file too.
Let's discuss this on Monday.

Reply:
In addition, what breaks a nice design a bit is that `date_time_modem.c` calls `date_time_store()` from `date_time_helpers.c` and then the helpers file calls `date_time_modem` in `date_time_store`.

Reply:
There is now `date_time_core.c` which has most of the core functionality. `date_time.c` still has API and some of its functionality whenever it doesn't need any state information. If state information is needed, it'll call `date_time_core.c`.